cachi2/core/package_managers/yarn/project.py

```diff
@@ -91,11 +97,21 @@ def package_manager(self) -> Optional[str]:
         """Get the package manager string."""
         return NotImplemented
 
+    @package_manager.setter
+    def package_manager(self, package_manager: str) -> None:
```

Comment:
Not sure how far we want to take the concept of immutability, but an alternative to this setter is adding a utility method such as [this](https://github.com/brunoapimentel/cachi2/commit/6879b1e5ea56cdaf0fa341cfd7ce51df347b244d#diff-482c5a124171fdd52977104dd909f1a274ec6debcd844ea4f60ba03f37d21de5R89-R93).

Reply:
That's a good point. We could disassociate the concept of `_path` from the class, and have all in-memory representations of the file that could be written to any path. But at this point, I think let's just make it mutable. :sweat_smile: